internal/controllers/app_controller.go

```diff
@@ -275,6 +291,12 @@ func checkPodStatus(c client.Client, appName string, depVersion ketchv1.Deployme
 		if pod.Status.Phase != v1.PodRunning {
 			return errors.New("all pods are not running")
 		}
+
+		for _, c := range pod.Status.Conditions {
```

Comment:
what if an application consists of several containers and one of them has a script that performs an action and exits.